### ISSUE TITLE
pypi.password is literally "<hidden>" in the web request

### ISSUE THREAD

**Issue by ytimenkov** [pdm-project/pdm#1905]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce
1. Configure custom PyPI index which requires authentication (we run Artifactory with a virtual repo)
```console
$ pdm config pypi.url https://noid.jfrog.io/artifactory/api/pypi/python-all/simple
$ pdm config pypi.username ytimenkov
$ pdm config pypi.password anytoken
```

2. Try installing any project (`pdm install`).

## Actual behavior
Log says:
```
unearth.collector: Failed to collect links from https://noid.jfrog.io/artifactory/api/pypi/noid-python-all/simple/pulumi/: Client Error(403): 
```

## Expected behavior
PDM uses proper token

## Environment Information

```bash
PDM version:
  2.5.6
Python Interpreter:
  C:\Users\Yuri\noid\infra-ci\.venv\Scripts\python.exe (3.11)
Project Root:
  C:/Users/Yuri/noid/infra-ci
Local Packages:
  
{
  "implementation_name": "cpython",
  "implementation_version": "3.11.3",
  "os_name": "nt",
  "platform_machine": "AMD64",
  "platform_release": "10",
  "platform_system": "Windows",
  "platform_version": "10.0.22621",
  "python_full_version": "3.11.3",
  "platform_python_implementation": "CPython",
  "python_version": "3.11",
  "sys_platform": "win32"
}
```

## Additional info
I tried to debug the issue and I get this header in the request: `'Authorization': 'Basic eXVyaS50aW1lbmtvdkBub2lkZ2FtZXMuY29tOjxoaWRkZW4+'` (which is literally "hidden").

**Comment by frostming** [comment]
This has been fixed in 2.6.0.

**Comment by ytimenkov** [comment]
Lovely!
